File: Samples/azure-storage/Azure.Python/storagemanagementclient/models/resource.py
# coding=utf-8
# --------------------------------------------------------------------------
# --------------------------------------------------------------------------

from msrest.serialization import Model


class Resource(Model):
    """Resource

    :param id: Resource Id
    :type id: str
    :param name: Resource name
    :type name: str
    :param type: Resource type
    :type type: str
    :param location: Resource location
    :type location: str
    :param tags: Resource tags
    :type tags: dict
    """ 

    _validation = {
        '_id': {'readonly': True},
        '_name': {'readonly': True},
        '_type': {'readonly': True},
    }

    _attribute_map = {
        '_id': {'key': 'id', 'type': 'str'},
        '_name': {'key': 'name', 'type': 'str'},
        '_type': {'key': 'type', 'type': 'str'},
        'location': {'key': 'location', 'type': 'str'},
        'tags': {'key': 'tags', 'type': '{str}'},
    }

    def __init__(self, location=None, tags=None):
        self._id = None
        self._name = None
        self._type = None
        self.location = location
        self.tags = tags

    @property
    def id(self):
        return self._id

    @property
    def name(self):
        return self._name

    @property
    def type(self):
        return self._type

File: src/client/Ruby/ms-rest-azure/lib/ms_rest_azure/resource.rb
# encoding: utf-8
# Copyright (c) Microsoft Corporation. All rights reserved.
# Licensed under the MIT License. See License.txt in the project root for license information.

module MsRestAzure
  #
  # Class which represents any Azure resource.
  #
  class Resource

    # @return [String] the id of the resource.
    attr_accessor :id

    # @return [String] the name of the resource.
    attr_accessor :name

    # @return [String] the type of the resource.
    attr_accessor :type

    # @return [String] the location of the resource (required).
    attr_accessor :location

    # @return [Hash{String => String}] the tags attached to resources (optional).
    attr_accessor :tags

    # @return [String] the kind of the resource.
    # FIXME kind attribute was introduced because of SiteInstance model of azure_mgmt_web gem has kind attribute.
    attr_accessor :kind

    def self.mapper
      {
        required: false,
        serialized_name: 'Resource',
        type: {
          name: 'Composite',
          class_name: 'Resource',
          model_properties: {
              id: {
                required: false,
                serialized_name: 'id',
                type: {
                  name: 'String'
                }
              },
              name: {
                required: false,
                read_only: true,
                serialized_name: 'name',
                type: {
                  name: 'String'
                }
              },
              type: {
                required: false,
                read_only: true,
                serialized_name: 'type',
                type: {
                  name: 'String'
                }
              },
              location: {
                required: false,
                serialized_name: 'location',
                type: {
                  name: 'String'
                }
              },
              tags: {
                required: false,
                serialized_name: 'tags',
                type: {
                  name: 'Dictionary',
                  value: {
                      required: false,
                      serialized_name: 'StringElementType',
                      type: {
                        name: 'String'
                      }
                  }
                }
              },
              kind: {
                required: false,
                serialized_name: 'kind',
                type: {
                  name: 'String'
                }
              }
          }
        }
      }
    end
  end
end
